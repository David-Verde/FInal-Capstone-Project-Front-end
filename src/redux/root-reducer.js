import categoryReducer from './category/category.reducer';
import propertyReducer from './property/property.reducer';
import reservationReducer from './reservation/reservation.reducer';
import statusReducer from './status/status.reducer';
import userReducer from './user/user.reducer';

export const rootReducer = {
  user: userReducer,
  property: propertyReducer,
  category: categoryReducer,
  reservation: reservationReducer,
  status: statusReducer,
};
