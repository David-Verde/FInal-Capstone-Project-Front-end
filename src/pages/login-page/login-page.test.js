import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../../redux/store';
import LoginPage from './loginPage.component';

const renderWithProviders = (reduxStore, route) => render(
  <Provider store={reduxStore}>
    <MemoryRouter initialEntries={[route]}>
      <LoginPage />
    </MemoryRouter>
  </Provider>,
);

describe('LoginPage component', () => {
  it('should render correctly', () => {
    const component = renderWithProviders(store, '/');
    expect(component).toMatchSnapshot();
  });
});
