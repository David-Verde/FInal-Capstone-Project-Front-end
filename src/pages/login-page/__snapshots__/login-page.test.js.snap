// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sign-ex"
      >
        <div
          class="sign-ex__container"
        >
          <div
            class="sign-ex__container__header"
          >
            <h2>
              Welcome to Home Sharing
            </h2>
            <p>
              Reserve the weekend of your dreams!
            </p>
          </div>
          <form
            class="sign-ex__container__form"
          >
            <div
              class="sign-ex__container__form__inputBox"
            >
              <input
                name="email"
                placeholder=" "
                required=""
                type="email"
                value=""
              />
              <span>
                Email
              </span>
              <i />
            </div>
            <div
              class="sign-ex__container__form__inputBox"
            >
              <input
                name="password"
                placeholder=" "
                required=""
                type="password"
                value=""
              />
              <span>
                Password
              </span>
              <i />
            </div>
            <button
              class="sign-ex__container__form__btn"
              type="submit"
            >
              Log In
            </button>
          </form>
        </div>
        <p
          class="sign-ex__new"
        >
          Need an account?
          <a
            class="sign-ex__new__link"
            href="/sign-up"
          >
            Sign up
          </a>
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sign-ex"
    >
      <div
        class="sign-ex__container"
      >
        <div
          class="sign-ex__container__header"
        >
          <h2>
            Welcome to Home Sharing
          </h2>
          <p>
            Reserve the weekend of your dreams!
          </p>
        </div>
        <form
          class="sign-ex__container__form"
        >
          <div
            class="sign-ex__container__form__inputBox"
          >
            <input
              name="email"
              placeholder=" "
              required=""
              type="email"
              value=""
            />
            <span>
              Email
            </span>
            <i />
          </div>
          <div
            class="sign-ex__container__form__inputBox"
          >
            <input
              name="password"
              placeholder=" "
              required=""
              type="password"
              value=""
            />
            <span>
              Password
            </span>
            <i />
          </div>
          <button
            class="sign-ex__container__form__btn"
            type="submit"
          >
            Log In
          </button>
        </form>
      </div>
      <p
        class="sign-ex__new"
      >
        Need an account?
        <a
          class="sign-ex__new__link"
          href="/sign-up"
        >
          Sign up
        </a>
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
